From 4cfc126893fac78a71a34589386f60de20d1a571 Mon Sep 17 00:00:00 2001
From: Jeremy Linton <jeremy.linton@arm.com>
Date: Wed, 14 Sep 2016 17:32:31 -0500
Subject: [PATCH 27/56] arm64: pmu: Hoist pmu platform device name

Git-commit: 85023b2e1325826edf5d226a9cb4d809ed1e2024
Patch-mainline: v4.9-rc1
References: bsc#1062279

Move the PMU name into a common header file so it may
be referenced by other users.

Signed-off-by: Jeremy Linton <jeremy.linton@arm.com>
Signed-off-by: Will Deacon <will.deacon@arm.com>
Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 arch/arm64/kernel/perf_event.c |    2 +-
 include/linux/perf/arm_pmu.h   |    2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

--- a/arch/arm64/kernel/perf_event.c
+++ b/arch/arm64/kernel/perf_event.c
@@ -1111,7 +1111,7 @@ static int armv8_pmu_device_probe(struct
 
 static struct platform_driver armv8_pmu_driver = {
 	.driver		= {
-		.name	= "armv8-pmu",
+		.name	= ARMV8_PMU_PDEV_NAME,
 		.of_match_table = armv8_pmu_of_device_ids,
 		.suppress_bind_attrs = true,
 	},
--- a/include/linux/perf/arm_pmu.h
+++ b/include/linux/perf/arm_pmu.h
@@ -159,6 +159,8 @@ int arm_pmu_device_probe(struct platform
 			 const struct of_device_id *of_table,
 			 const struct pmu_probe_info *probe_table);
 
+#define ARMV8_PMU_PDEV_NAME "armv8-pmu"
+
 #endif /* CONFIG_ARM_PMU */
 
 #endif /* __ARM_PMU_H__ */
